src/legacy/core_plugins/kibana/public/discover/np_ready/angular/context_state.ts

```diff
@@ -126,7 +126,7 @@ export function getState({
 }: GetStateParams): GetStateReturn {
   const stateStorage = createKbnUrlStateStorage({
     useHash: storeInSessionStorage,
-    history: history ? history : createBrowserHistory(),
+    history: history ? history : createHashHistory(),
```

Comment:
I would get rid of `createHashHistory` at all since `getState` is not used without history passing (and adjust type)
```suggestion
    history,
```